src/libostree/ostree-repo-pull.c

```diff
@@ -1322,6 +1324,165 @@ commitstate_is_partial (OtPullData   *pull_data,
+  additional_metadata = g_variant_get_child_value (summary, 1);
+
+  const char *main_collection_id;
+  if (!g_variant_lookup (additional_metadata,
+                         OSTREE_SUMMARY_COLLECTION_ID,
+                         "&s",
+                         &main_collection_id))
+    main_collection_id = NULL;
+
+  gboolean check_main_collection;
+  check_main_collection = (g_strcmp0 (main_collection_id, collection_id) == 0);
+
+  if (check_main_collection)
+    {
+      g_autoptr(GVariant) refs = NULL;
+      gsize i, n;
+
+      refs = g_variant_get_child_value (summary, 0);
+      for (i = 0, n = g_variant_n_children (refs); i < n; i++)
+        {
+          const char *refname;
+          g_autoptr(GVariant) ref_v = g_variant_get_child_value (refs, i);
+
+          g_variant_get_child (ref_v, 0, "&s", &refname);
+          if (g_strcmp0 (refname, ref) == 0)
+            return TRUE;
+        }
+
+      return FALSE;
+    }
+
+  g_autoptr(GVariant) collection_map = NULL;
+  collection_map = g_variant_lookup_value (additional_metadata,
+                                           OSTREE_SUMMARY_COLLECTION_MAP,
+                                           G_VARIANT_TYPE ("a{sa(s(taya{sv}))}"));
+  if (collection_map != NULL)
+    {
+      GVariantIter collection_map_iter;
+      char *unpack_summary_collection_id;
+      GVariant *unpack_refs;
+
+      g_variant_iter_init (&collection_map_iter, collection_map);
+
+      while (g_variant_iter_next (&collection_map_iter,
+                                  "{&s@a(s(taya{sv}))}",
+                                  &unpack_summary_collection_id,
+                                  &unpack_refs))
+        {
+          gsize i, n;
+          g_autofree char *summary_collection_id = unpack_summary_collection_id;
+          g_autoptr(GVariant) refs = unpack_refs;
+
+          if (g_strcmp0 (summary_collection_id, collection_id) != 0)
+            continue;
+
+          for (i = 0, n = g_variant_n_children (refs); i < n; i++)
+            {
+              const char *refname;
+              g_autoptr(GVariant) ref_v = g_variant_get_child_value (refs, i);
+
+              g_variant_get_child (ref_v, 0, "&s", &refname);
+
+              if (g_strcmp0 (refname, ref) == 0)
+                  return TRUE;
+            }
+        }
+    }
+
+  return FALSE;
+}
+
+static char *
+get_remote_collection_id (OstreeRepo  *repo,
+                          const gchar *remote_name)
+{
+  g_autofree gchar *remote_collection_id = NULL;
+
+  /* TODO: This happens for local pulls, I guess we should assume no
+   * collection id for this case?
+   */
+  if (remote_name == NULL)
+    return NULL;
+
+  if (!ostree_repo_get_remote_option (repo, remote_name, "collection-id", NULL,
+                                      &remote_collection_id, NULL) ||
+      remote_collection_id == NULL)
+    return NULL;
+
+  return g_steal_pointer (&remote_collection_id);
+}
+
+static gboolean
+verify_collection_ref (OtPullData  *pull_data,
```

Comment:
It would be useful to add a short doc comment for this function explaining what’s being verified and when verification is skipped.